WalletWasabi.Backend/Controllers/BlockchainController.cs

```diff
@@ -416,4 +417,127 @@ private async Task<StatusResponse> FetchStatusAsync(CancellationToken cancellati
 
 		return status;
 	}
+
+	[HttpGet("unconfirmed-transaction-chain")]
+	[ProducesResponseType(200)]
+	[ProducesResponseType(400)]
+	public async Task<List<UnconfirmedTransactionChainItem>> GetUnconfirmedTransactionChainAsync([FromQuery, Required] string transactionId, CancellationToken cancellationToken)
+	{
+		uint256 txId = new(transactionId);
+
+		var cacheKey = $"{nameof(GetUnconfirmedTransactionChainAsync)}_{txId}";
+		var cacheOptions = new MemoryCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(3) };
+
+		return await Cache.GetCachedResponseAsync(
+			cacheKey,
+			action: (string request, CancellationToken token) => GetUnconfirmedTransactionChainNoCacheAsync(txId, token),
+			options: cacheOptions,
+			cancellationToken);
+	}
+
+	private async Task<List<UnconfirmedTransactionChainItem>> GetUnconfirmedTransactionChainNoCacheAsync(uint256 txId, CancellationToken cancellationToken)
+	{
+		Dictionary<uint256, Transaction> transactionsLocalCache = new();
+		Dictionary<uint256, UnconfirmedTransactionChainItem> unconfirmedTxsChainById = new();
+		var mempool = Global.HostedServices.Get<MempoolMirror>();
+		var mempoolHashes = mempool.GetMempoolHashes();
+
+		if (!mempoolHashes.Contains(txId))
+		{
+			throw new InvalidOperationException("Requested transaction is not present in the mempool, probably confirmed.");
+		}
+
+		// TODO: Use Transaction cache.
+		var requestedTransaction = await RpcClient.GetRawTransactionAsync(txId, true, cancellationToken);
+
+		transactionsLocalCache.Add(txId, requestedTransaction);
+
+		List<Transaction> toFetchFeeList = new() { requestedTransaction };
+
+		while (toFetchFeeList.Count > 0)
+		{
+			var currentTx = toFetchFeeList.First();
+
+			List<Coin> inputs = new();
+			HashSet<Transaction> parentTxs = new();
+
+			void AddParentTx(Transaction tx, OutPoint prevOut)
+			{
+				parentTxs.Add(tx);
+				TxOut txOut = tx.Outputs[prevOut.N];
+				inputs.Add(new Coin(prevOut, txOut));
+			}
+
+			// Collect which transactions needs to be fetched.
+			List<OutPoint> prevOutToFetchFromRPC = new();
+
+			foreach (var input in currentTx.Inputs)
+			{
+				if (transactionsLocalCache.TryGetValue(input.PrevOut.Hash, out var parentTx))
+				{
+					AddParentTx(parentTx, input.PrevOut);
+				}
+				else
+				{
+					prevOutToFetchFromRPC.Add(input.PrevOut);
+				}
+			}
+
+			if (prevOutToFetchFromRPC.Count > 0)
+			{
+				var missingTxs = prevOutToFetchFromRPC.Count == 1 ?
+					new [] { await RpcClient.GetRawTransactionAsync(prevOutToFetchFromRPC.Single().Hash, false, cancellationToken) } :
```

Comment:
> Honestly, I'm not sure that this is required. I wrote it because you made it, but I'm not sure that there is any problem to call GetRawTransactionsAsync with a single transaction

No, I don't think there is a problem with calling it that way.

> Also, I changed throwIfNotFound to false because I'm handling this case right below

With this change, I see no reason to call it two different way, when the underlying methods will do call the same method with the same parameters.